src/services/files/file.hooks.js

```diff
@@ -73,11 +75,86 @@ const hasViewPermission = hook => {
     });
 };
 
+const getCurrentUserData = hook => {
+  const userModel = hook.app.get('mongooseClient').model('users');
+    return new Promise((resolve, reject) => {
+      userModel.findById(hook.params.user._id, function (err, user) {
+        if(err) { reject(new errors.GeneralError(err)); }
+        if(!user) {
+          reject(new errors.NotFound('User not found'));
+        }
+        hook.params.user.role = user.role;
+        hook.params.user.providerId = user.providerId;
+        return resolve(hook);
+      });
+    });
+};
+
+const restrictResourceToCurrentProvider = async hook => {
+  if(hook.params.user.role !== 'superhero') {
+    const resource = await hook.app.service('resources').get(hook.id);
+    if(!(resource.providerId == hook.params.user.providerId.toString())) {
+      throw new errors.Forbidden('Permissions missing');
+    }
+  }
+  return hook;
+};
+
+const restrictToProviderOrToPublicResources = hook => {
+  if(hook.params.user.role !== 'superhero') {
+    if(!(hook.params.resource.providerId.toString() === hook.params.user.providerId.toString() || hook.params.resource.isPublished === true)) {
+      throw new errors.Forbidden('Permissions missing');
+    }
+  }
+  return hook;
+};
+
+const restrictToPublicResources = hook => {
+  if(hook.params.resource.isPublished != true) {
+    throw new errors.Forbidden('Permissions missing');
+  }
+  return hook;
+};
+
+const getResource = async hook => {
+  const path = hook.params.route[0];
+  let [resourceId] = unifySlashes(path)
+    .replace(/^\/+/g, '') // remove leading slashes
+    .split('/');
+  const resource = await hook.app.service('resources').get(resourceId);
+  hook.params.resource = resource;
+  return hook;
+};
+
+// Lern-Store can only access public resources, authenticated users can access public resources and resources of their company
+const checkPermissionsAfterAuthentication = hook => {
```

Comment:
You must also implment the auth token logic you removed here: https://github.com/schul-cloud/schulcloud-content/pull/69/files#diff-d66f0b421e3e976145d1baff79f60b14R21
Also notice that you need to remove the unused method referenced above, after you implemented the logic.

Reply:
We have a ticket for this and solve it later. https://ticketsystem.schul-cloud.org/browse/BALS-58